### PR TITLE
Addressed the first three TODOs

### PR DESCRIPTION
**Pull request by photex**
Howdy!

The first three TODO tickets were addressed in this PR, but I haven't yet been able to get some things working otherwise. In particular, updating packages fails on my setup due to an exception in the spacewalk python modules:
AttributeError: dbPackage instance has no attribute 'nvrea'
Also my spacewalk server (installed from nightlies) doesn't have a salt.job concept. I need to try and rebuild my server presumably from the repo hosted here on github before I can address the remaining items.

### CODE REVIEW

**Comment by photex** [comment]
Realized the issue with package additions and have pushed the fix for that. The extended package information is still bogus while I determine how to get this information with salt. The salt docs make reference to a pkg.info command but that isn't available afaict with 2014.7.5.

**Comment by dmacvicar** [comment]
Hi!

I was thinking, probably you are missing the database part we changed in Spacewalk. I think I forgot to move that from the internal branch.

schema/spacewalk/common/tables/rhnSaltClient.sql
```
CREATE TABLE rhnSaltClient
(
    server_id          NUMBER NOT NULL,
    minion_id          VARCHAR2(128),
    created            timestamp with local time zone
                           DEFAULT (current_timestamp) NOT NULL,
    modified           timestamp with local time zone
                           DEFAULT (current_timestamp) NOT NULL
)
ENABLE ROW MOVEMENT
;

CREATE UNIQUE INDEX rhn_saltclient_minion_id_uq
    ON rhnSaltClient (minion_id)
    TABLESPACE [[8m_tbs]];

CREATE UNIQUE INDEX rhn_saltclient_sid_uq
    ON rhnSaltClient (server_id)
    TABLESPACE [[2m_tbs]];
```

If you have any suggestion about this sql snippet... as it does not belong in this repository but in Spacewalk, we can keep it in the README for now, or some Makefile that given the spacewalk server injects it for you.

**Comment by photex** [comment]
Good morning!

I don't see those tables referenced in salt-registerd so I need to look deeper at the registration to make sure I understand where those fit in.

The other table missing is 'rhnActionSaltJob'. Also, when scheduling the action the action type 'salt.job' causes an exception be raised as it's not a known type. Is that defined in the schema or is it data that I can insert?

I chose the path of letting this script do as much as it can and clearly report the problems it finds, but perhaps a check at startup would be preferrable instead (exiting early). Especially if there is another script or Makefile that can update Spacewalk. My initial preference would be to create a folder 'setup' with another README.md with instructions on how to patch Spacewalk. 

The way that Spacewalk deals with SQL is a bit foggy for me still so I've cloned the repo and will spend some time looking into it (sql scripts in common are processed somehow into something more specific for the backend you're running?).

**Comment by dmacvicar** [comment]
Yes, when we opened the code there is a big chunk of code that was patched to SUSE Manager. I would need to rebase that into Spacewalk, but that would mean you would need to rebuild Spacewalk :-)

File: bin/salt-registerd [hunk 1]
```diff
@@ -183,93 +223,276 @@ class Saltwalk:
...
+def process_package_list_result(minion, event):
+    """
+    Updates the server registration of the minion with
+    the list of packages reported by salt.
+    """
+    logger.info('Got package list for %s', minion)
+
+    sid = is_minion_registered(minion)
+
+    package_list = format_package_list(event['data']['return'])
```
Comment (isbm) [inline bin/salt-registerd]: Maybe this makes sense to be executed only if `sid` is around? I would move this below, like:
```python
for package in format_package_list(event.get('data', {}).get('return', {})):
    server.add_package(.....
```

File: bin/salt-registerd [hunk 2]
```diff
@@ -3,67 +3,132 @@
...
+        rhnSQL.rollback()
+
+    try:
+        yield
+        rhnSQL.commit()
+    except rhnFault as fault:
+        logger.exception("Server fault caught: %s", fault)
+        rollback()
+    except Exception as error:
+        msg = error.message
```
Comment (isbm) [inline bin/salt-registerd]: This is deprecated. It is enough to just `str(error)` for these purposes.

File: bin/salt-registerd [hunk 3]
```diff
@@ -73,106 +138,81 @@ def register_system(data):
...
+    """
+    Returns sid if a minion is registered, False otherwise
+    """
+    result = False
+    cursor = rhnSQL.prepare("""
+    SELECT id from rhnServer
+    WHERE digital_server_id=:did
+    """)
+    query_results = cursor.execute(did=('SALT-ID-%s' % minion))
+    if query_results > 0:
```
Comment (isbm) [inline bin/salt-registerd]: Unless we are looking here for negative codes, I wouldn't do that. DBI does not defines the return codes and cx_Oracle returns `None` here, for example. Maybe it is enough to just:
```python
if cursor.execute(did=('SALT-ID-%s' % minion)):
    result = cursor.fetchone_dict().get('id', False)
return result
```

**Comment by isbm** [comment]
@photex Hi! Overall, cleanup looks very nice. I agree that it makes sense to go forward and move functions to the classes. Thanks for the contribution!

**Comment by photex** [comment]
Thanks for the comments @isbm. I'll get a proper context manager put together next.

**Comment by isbm** [comment]
@photex I also suggest to _not_ use string substitutions, but a named formatting instead. It is easier for non-pythoneers to get where goes what. :smiley: But this is up to you.

**Comment by photex** [comment]
@isbm ya know it's just a force of an old habit and not a real preference. I actually prefer to see named formatting myself so I should tidy that up as well. :)

**Comment by isbm** [comment]
Old habits die hard. Looking forward for new PRs, @photex!

**Pull request merged**